src/txmempool.cpp

```diff
@@ -752,3 +761,27 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
     // Estimate the overhead of mapTx to be 6 pointers + an allocation, as no exact formula for boost::multi_index_contained is implemented.
     return memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 6 * sizeof(void*)) * mapTx.size() + memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) + cachedInnerUsage;
 }
+
+void CTxMemPool::setMempoolCostLimit(int64_t totalCostLimit, int64_t evictionMemorySeconds) {
+    LogPrint("mempool", "Setting mempool cost limit: (limit=%d, time=%d)\n", totalCostLimit, evictionMemorySeconds);
+    // This method should not be called more than once
+    assert(!recentlyEvicted);
+    assert(!weightedTxList);
+    recentlyEvicted = new RecentlyEvictedList(evictionMemorySeconds);
+    weightedTxList = new WeightedTransactionList(totalCostLimit);
```

Comment:
If we are storing these as pointers, we need to explicitly free them in `CTxMemPool::~CTxMemPool`.